File: packages/types/typescript/test_translator.ts
import { enumTranslator } from "./translateEnums";
import { RacePhase, Item, NumCoins } from "./generated/data";

function assert(condition: boolean, message: string) {
  if (!condition) {
    throw new Error(`Assertion failed: ${message}`);
  }
}

console.log("Running TypeScript Translator Tests...");

// Test RacePhase
assert(enumTranslator.racePhaseEnumToString(RacePhase.RACING) === "racing", "RacePhase to String");
assert(enumTranslator.racePhaseStringToEnum("racing") === RacePhase.RACING, "String to RacePhase");
assert(
  enumTranslator.racePhaseStringToEnum("RACING") === RacePhase.RACING,
  "Case insensitive RacePhase"
);

// Test Item
assert(enumTranslator.itemEnumToString(Item.GREEN_SHELL) === "green_shell", "Item to String");
assert(enumTranslator.itemStringToEnum("green_shell") === Item.GREEN_SHELL, "String to Item");
assert(
  enumTranslator.itemStringToEnum("triple_red_shells") === Item.TRIPLE_RED_SHELLS,
  "Triple shells"
);

// Test NumCoins
assert(enumTranslator.numCoinsEnumToString(NumCoins.COIN_10) === "coin_10", "Coins to String");
assert(enumTranslator.numCoinsStringToEnum("coin_10") === NumCoins.COIN_10, "String to Coins");

console.log("✅ All TypeScript tests passed!");

File: scratch/chase/streamlit/utils/packages/types/python/test_translator.py
import sys
import os
import unittest

# Add parent directory to path to find the package
sys.path.append(os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))))

from packages.types.python.translate_enums import enumTranslator
import packages.types.python.generated.data_pb2 as data

class TestEnumTranslator(unittest.TestCase):
    def test_race_phase(self):
        self.assertEqual(enumTranslator.racePhaseEnumToString(data.RACING), "racing")
        self.assertEqual(enumTranslator.racePhaseStringToEnum("racing"), data.RACING)
        self.assertEqual(enumTranslator.racePhaseEnumToString(999), "phase_unknown")

    def test_item(self):
        self.assertEqual(enumTranslator.itemEnumToString(data.GREEN_SHELL), "green_shell")
        self.assertEqual(enumTranslator.itemStringToEnum("green_shell"), data.GREEN_SHELL)
        self.assertEqual(enumTranslator.itemStringToEnum("triple_red_shells"), data.TRIPLE_RED_SHELLS)

    def test_coins(self):
        self.assertEqual(enumTranslator.numCoinsEnumToString(data.COIN_10), "coin_10")
        self.assertEqual(enumTranslator.numCoinsStringToEnum("coin_10"), data.COIN_10)

    def test_case_insensitivity(self):
        self.assertEqual(enumTranslator.racePhaseStringToEnum("RaCiNg"), data.RACING)

if __name__ == "__main__":
    unittest.main()
